src/sentry/apidocs/pagination_linter.py

```diff
@@ -0,0 +1,75 @@
+import ast
+
+
+class FindPaginateInGetMethodVisitor(ast.NodeVisitor):
+    def __init__(self, class_name):
+        self.class_name = class_name
+        self.function_defs = {}  # Maps function names to their AST nodes
+        self.called_by_get = []  # List of functions called directly by 'get'
+        self.visited = set()  # Prevent infinite recursion
+        self.found = False
+
+    def visit_ClassDef(self, node):
+        if node.name == self.class_name:
+            # Pre-process to map function names to their nodes
+            for n in node.body:
+                if isinstance(n, ast.FunctionDef):
+                    self.function_defs[n.name] = n
+            self.generic_visit(node)
+
+    def visit_FunctionDef(self, node):
+        if node.name == "get":
+            self.generic_visit(node)  # Visit to find all direct calls
+            # Now recursively visit all functions called by 'get'
+            for func_name in self.called_by_get:
+                self.visit_function_calls(func_name)
+        elif node.name in self.function_defs:  # Prevent revisiting from the class level
+            super().generic_visit(node)  # Normal traversal for other methods
+
+    def visit_Call(self, node):
+        if (
+            isinstance(node.func, ast.Attribute)
+            and node.func.attr == "paginate"
+            and isinstance(node.func.value, ast.Name)
+            and node.func.value.id == "self"
+        ):
+            self.found = True
```

Comment:
if we have this same logic in `visit_FunctionDef` -> `visit_function_calls`, do we need a special method for visiting `Calls` here?

Reply:
There is two cases where `self.paginate` could exist - directly within the `get` method and in some other function. (either nested or not). `visit_Call` detects `self.paginate` when its directly in the `get` method.
`visit_FunctionDef` allows us to detect the paginate when its in other functions that indirectly call paginate.

Reply:
i see